docs/DeveloperGuide.md

```diff
@@ -4,10 +4,220 @@
+
+Step 2. The user executes `appointment add s/TRIM p/DOGGO d/20-10-2022` command to add an appointment for trimming service for ‘DOGGO’ on 20-10-2022. This updates `AppointmentList` with the appointment.
+
+Step 3. The user executes `appointment status i/2 s/COMPLETED` command to update the status of the above appointment to completed. This removes the appointment from `AppointmentList` and adds the appointment to `AppointmentHistory`.
+
+Step 4. The user executes `appointment history` command to view the list of expired appointment. This will display the list of appointments that has already been completed by the clinic. Then, the user ends with application with `bye` command, the current state of `AppointmentList` and `AppointmentHistory` will be stored in a file through `Storage` .
+
+> The following activity diagram summarizes what happens when a user executes a new command:
+
+### Design considerations:
+
+**Aspect: How undo & redo executes:**
+
+- Alternative 1 (current choice):
+
+   Saves the entire address book.
+
+  - Pros: Easy to implement.
+  - Cons: May have performance issues in terms of memory usage.
+
+- Alternative 2:
+
+   Individual command knows how to undo/redo by itself.
+
+  - Pros: Will use less memory (e.g. for `delete`, just save the person being deleted).
+  - Cons: We must ensure that the implementation of each individual command are correct.
+
+*{more aspects and alternatives to be added}*
+
+### [Proposed] Employee management feature
+
+### Proposed Implementation
+
+The proposed employee management mechanism is facilitated by `Employee` , `EmployeeList`, `Storage`. It implements the following operations:
+
+- `EmployeeList#addEmployee()` — Adds an employee to the employee list.
+- `EmployeeList#listEmployee()` — Views all the employees in the employee list.
+- `EmployeeList#removeEmployee()` — Removes an employee in the employee list.
+- `Storage#loadEmployee()` —  Storage stores the list of current employees into a file and this function loads it when application starts.
+
+Given below is an example usage scenario and how the employee management mechanism behaves at each step.
+
+Step 1. The user launches the application and there are already pre-existing employees loaded. `EmployeeList` initializes and loads from `Storage` the existing employees.
+
+Step 2. The user executes `employee add n/Mozart` command to add an employee named Mozart. This updates `EmployeeList` with the employee.
+
+Step 3. The user executes `employee view` command to view the current employees. 
+
+Step 4. The user executes `employee remove i/1` command to remove the employee with index 1 from the employee list.
+
+The following sequence diagram shows how the employee management works:
+
+![https://github.com/AY2223S1-CS2113-F11-2/tp/blob/master/docs/uml/EmployeeManagement.png](https://github.com/AY2223S1-CS2113-F11-2/tp/blob/master/docs/uml/EmployeeManagement.png)
```

Comment:
The first diagram here seems to contradicts with the diagram found at a lower portion of your DG. The second diagram seems to infer that the input is parsed after the Command class but this contradicts with the first diagram which does not parse the input `employee add n/Mozart`
 
![image](https://user-images.githubusercontent.com/37980219/197934272-d0df2528-a242-436d-8955-e6bfaa199de6.png)
![image](https://user-images.githubusercontent.com/37980219/197934382-52cbdbed-fb27-45d6-8c2e-0a632b2d8ef3.png)